src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -44,7 +47,11 @@ public static int boundedNumberOfProcessors(Settings settings) {
          * ie. >= 48 create too many threads and run into OOM see #3478
          * We just use an 32 core upper-bound here to not stress the system
          * too much with too many created threads */
-        return settings.getAsInt(PROCESSORS, Math.min(32, Runtime.getRuntime().availableProcessors()));
+        int defaultValue = Math.min(32, Runtime.getRuntime().availableProcessors());
```

Comment:
oh nevermind I didn't see what we still use the settings in the return statement.... all is well